commit 5f32083c759b468f5b0cb9302e4fe2848d37cc5b
From: Steve Dickson <steved@redhat.com>
Date: Tue, 10 Apr 2018 10:25:40 -0400
Subject: mount.nfs: Fix auto protocol negotiation
 Commit 71b807e1 introduce a regression that
 caused v3 not to be tried when v4 was not
 supported by the server during auto negation.
 
 A check of the type in nfs_nfs_version() was
 reverted back to only check for the "nfs4"
 string not the "nfs" string which fixed the
 problem.
 
 Signed-off-by: Steve Dickson <steved@redhat.com>
Bug-Ubuntu: https://bugs.launchpad.net/bugs/2059197
Bug: https://www.spinics.net/lists/linux-nfs/msg68073.html
Origin: upstream, https://git.linux-nfs.org/?p=steved/nfs-utils.git;a=commitdiff;h=5f32083c759b468f5b0cb9302e4fe2848d37cc5b
Last-Update: 2024-04-11

Index: nfs-utils-1.3.4/utils/mount/network.c
===================================================================
--- nfs-utils-1.3.4.orig/utils/mount/network.c	2024-04-11 13:29:18.201973577 +1200
+++ nfs-utils-1.3.4/utils/mount/network.c	2024-04-11 13:29:18.197973522 +1200
@@ -1266,8 +1266,8 @@
 		}
 	}
 
-	if (!found && strncmp(type, "nfs", 3) == 0)
-		version_val = "4";
+	if (!found && strcmp(type, "nfs4") == 0)
+		version_val = type + 3;
 	else if (!found)
 		return 1;
 	else if (i <= 2 ) {
